lib/user_stats.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require_relative 'bus'
+
+module UserStats
+  K = 40.0
+  K_NEW = 60.0
+  YEARS_INCLUDED = 4
+  BATCH_SIZE = 1000
+
+  def self.calculate_stats
+    user_stats = Hash.new { |hash, key| hash[key] = Hash.new([1200, 0]) }
+    on_or_after = Date.today - (365.25 * YEARS_INCLUDED)
+
+    Game.eager(:players).where(status: %w[finished archived], manually_ended: false)
+                        .where { finished_at >= on_or_after }
+                        .order(:finished_at)
+                        .paged_each(rows_per_fetch: BATCH_SIZE) do |game|
+      calculate_game_stats(game, user_stats)
+    end
+
+    store_stats(user_stats)
+    true
```

Comment:
why return true?

Reply:
Before it was returning the user_stats hash, which is long. What would you like it to return?

Reply:
well if you don't need it to return anything don't return anything

Reply:
So put a nil at the bottom?

Reply:
no, just don't do anything

Reply:
Ruby will return the output of the last line of the method. What's your opinion on what to do if that last line is a really big hash, string, etc.?

Reply:
it's fine, just don't use it, you should only return stuff purposefully if you're going to use the value